### PR TITLE
Various ili9341 fixes

### PR DESCRIPTION
**Pull request by nielsnl68**
- display_() crashed when noting needed to be displayed.
- simplyfied the fill_internal_() and fill buffer_ and screen same color
- the update area is now only being updated when the pixel changed
- fixed initiation clear screen where only half the screen was cleared

# What does this implement/fix?

Quick description and explanation of changes

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/ili9341/ili9341_display.cpp [hunk 1]
```diff
@@ -88,10 +86,19 @@ void ILI9341Display::display_() {
...
+
+  // check if something was displayed
+  if ((this->x_high_ < this->x_low_) || (this->y_high_ < this->y_low_)) {
+    return;
+  }
 
   set_addr_window_(this->x_low_, this->y_low_, w, h);
+
+  ESP_LOGD("ILI9341", "Start ILI9341Display::display_(xl:%d, xh:%d, yl:%d, yh:%d, w:%d, h:%d, start_pos:%d)",
+           this->x_low_, this->x_high_, this->y_low_, this->y_high_, w, h, start_pos);
```
Comment (jesserockz) [inline esphome/components/ili9341/ili9341_display.cpp]: Remove or change to V/VV please

File: esphome/components/ili9341/ili9341_display.cpp [hunk 2]
```diff
@@ -101,9 +108,12 @@ void ILI9341Display::display_() {
       this->write_array(transfer_buffer_, 2 * sz);
       pos += sz;
       rem -= sz;
+      App.feed_wdt();
     }
+    App.feed_wdt();
   }
   this->end_data_();
+  ESP_LOGD("ILI9341", "Finish ILI9341Display::display_()");
```
Comment (jesserockz) [inline esphome/components/ili9341/ili9341_display.cpp]: Same here

File: esphome/components/ili9341/ili9341_display.cpp [hunk 3]
```diff
@@ -147,26 +147,58 @@ void ILI9341Display::fill_internal_(Color color) {
 
   this->end_data_();
 
-  memset(buffer_, 0, (this->get_width_internal()) * (this->get_height_internal()));
+  memset(buffer_, color, this->get_buffer_length_());
+}
+
+void ILI9341Display::rotate_my_(uint8_t m) {
```
Comment (jesserockz) [inline esphome/components/ili9341/ili9341_display.cpp:152]: This is not used anywhere?

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/ili9341/ili9341_display.cpp (hunk 3 above)
Comment (nielsnl68) [reply esphome/components/ili9341/ili9341_display.cpp:152]: Will implement it later, i used it to test my display, i have already so many changes in different components. 
that i found it to much to add this (breaking) changes as well.
I hope i can let it be until then?

**Review by jesserockz — APPROVED** [review]

**Pull request merged**